carball/analysis/stats/possessions.py

```diff
@@ -0,0 +1,200 @@
+from collections import Counter
+from typing import List, Dict, Tuple
+
+import pandas as pd
+
+from carball.analysis.stats.possession_based_stats import get_per_possession_stats
+from carball.generated.api.stats.events_pb2 import Hit
+from ...generated.api import game_pb2
+from ...generated.api.player_pb2 import Player
+from ...json_parser.game import Game
+
+
+def analyse_possessions(game: Game, proto_game: game_pb2.Game,
```

Comment:
This should use the existing class based methods and should be in the folder called possession rather than being free methods

Reply:
https://github.com/SaltieRL/carball/blob/master/carball/analysis/stats/possession/possession.py

For reference you can look at how these extend BaseStat and HitStat and choose the appropriate one